src/m365/spo/commands/list/list-roleinheritance-reset.spec.ts

```diff
@@ -0,0 +1,189 @@
+        containsTypeOption = true;
+      }
+    });
+    assert(containsTypeOption);
+  });
+
+  it('fails validation if both id and title options are not passed', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the url option is not a valid SharePoint site URL', () => {
+    const actual = command.validate({ options: { webUrl: 'foo' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the url option is a valid SharePoint site URL', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF' } });
+    assert(actual);
+  });
+
+  it('fails validation if the id option is not a valid GUID', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '12345' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if the id option is a valid GUID', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF' } });
+    assert(actual);
+  });
+
+  it('fails validation if both id and title options are passed', () => {
+    const actual = command.validate({ options: { webUrl: 'https://contoso.sharepoint.com', listId: '0CD891EF-AFCE-4E55-B836-FCE03286CCCF', listTitle: 'Documents' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('reset role inheritance on list by title', (done) => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/lists/getbytitle(\'test\)/resetroleinheritance') > -1) {
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        listTitle: 'test'
+      }
+    }, () => {
+      try {
+        assert(true);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('reset role inheritance on list by id', (done) => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if ((opts.url as string).indexOf('/_api/web/lists(guid\'202b8199-b9de-43fd-9737-7f213f51c991\')/resetroleinheritance') > -1) {
+        return Promise.resolve();
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        listId: '202b8199-b9de-43fd-9737-7f213f51c991'
+      }
+    }, () => {
+      try {
+        assert(true);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('list role inheritance reset command handles reject request correctly', (done) => {
+    const err = 'Invalid request';
```

Comment:
We should set this to a different string than what we have on line 170. Right now, this test should be failing because of a type in the request URL on line 166 (missing closing quote in list's title in the URL), but instead it's passing because in both cases we have the same error message.

Reply:
done